src/fu-util.c

```diff
@@ -2021,6 +2029,64 @@ fu_util_downgrade (FuUtilPrivate *priv, gchar **values, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_util_reinstall (FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	const gchar *remote_id;
+	FwupdRelease *rel = NULL;
+	g_autoptr(GPtrArray) rels = NULL;
+	g_autoptr(FwupdDevice) dev = fu_util_get_device_or_prompt (priv,
+								   values,
+								   error);
+	if (dev == NULL)
+		return FALSE;
+
+	/* try to lookup/match release from client */
+	rels = fwupd_client_get_releases (priv->client, fwupd_device_get_id (dev),
+					  NULL, error);
+	if (rels == NULL)
+		return FALSE;
+	for (guint j = 0; j < rels->len; j++) {
+		FwupdRelease *rel2 = g_ptr_array_index (rels, j);
+		if (fu_common_vercmp (fwupd_release_get_version (rel2),
+		    fu_device_get_version (dev)) == 0) {
+			rel = rel2;
+			break;
+		}
+	}
+	if (rel == NULL) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "Unable to locate release for %s version %s",
+			     fu_device_get_name (dev),
+			     fu_device_get_version (dev));
+		return FALSE;
+	}
+
+	/* update the console if composite devices are also updated */
+	priv->current_operation = FU_UTIL_OPERATION_INSTALL;
+	g_signal_connect (priv->client, "device-changed",
+			  G_CALLBACK (fu_util_update_device_changed_cb), priv);
+	priv->flags |= FWUPD_INSTALL_FLAG_ALLOW_REINSTALL;
+	if (!fu_util_update_device_with_release (priv, dev, rel, error))
+		return FALSE;
+	fu_util_display_current_message (priv);
+
+	/* send report if we're supposed to */
+	remote_id = fwupd_release_get_remote_id (rel);
+	if (!fu_util_maybe_send_reports (priv, remote_id, error))
```

Comment:
Semi-related question. Do you think reports are useful from version `n` to version `n`? Currently on the LVFS we just consider the target version, but I'm wondering if we want to filter the reports to be only upgrades server and client side?

Reply:
Well I think if someone is reinstalling something who isn't me or you it is interesting - it's a tell that they're trying to fix a problem that they think a reinstall would fix.

Reply:
Good point, agreed.